seonoh/ArchitectureStudy/app/src/main/java/com/example/seonoh/seonohapp/MainActivity.kt

```diff
@@ -14,28 +14,21 @@ class MainActivity : AppCompatActivity(), CoinMainContract.View {
     private lateinit var toast: Toast
     private lateinit var coinMarketNameList: List<String>
 
-    private val presenter: CoinMainPresenter by lazy {
-        CoinMainPresenter(
-            this
-        )
-    }
+    override val presenter: CoinMainContract.Presenter by lazy { CoinMainPresenter(this) }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
         setContentView(R.layout.activity_main)
         initView()
-        presenter.loadMarketData()
+        presenter.loadData()
     }
 
     private fun initView() {
         pagerAdapter = TabPagerAdapter(supportFragmentManager)
         coinViewPager.apply {
             adapter = pagerAdapter
-            addOnPageChangeListener(object : TabLayout.TabLayoutOnPageChangeListener(tabLayout) {
-                override fun onPageSelected(position: Int) {
-                    super.onPageSelected(position)
-                }
-            })
+            addOnPageChangeListener(object : TabLayout.TabLayoutOnPageChangeListener(tabLayout) {})
```

Comment:
오버라이딩하여 사용하지 않으실 것이라면, 이렇게 쓰시는게 더 좋아보이네요.
```suggestion
            addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(tabLayout))
```

Reply:
앗! 감사합니다.
5ed332fd9b3d31f0d0a24e6a456660a4bf1311fa